File: Coding/03_Beginner/Full_search/Java/12.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int A = sc.nextInt();
        int B = sc.nextInt();
        int N = Math.min(A, B);
        int ans = 0;

        for (int i = 1; i <= N; i++){
            if (A % i == 0 && B % i == 0) {
                ans = i;
            }
        }
    
    System.out.println(ans);
    }
}

File: Coding/03_Beginner/Full_search/JavaScript/12.js
const fs = require("fs");
const [A, B] = fs.readFileSync(0, "utf8").trim().split(" ").map(Number);

let N = Math.min(A, B);
let ans = 1;

for (let i = 1; i <= N; i++) {
  if (A % i === 0 && B % i === 0) {
    ans = i;
  }
}

console.log(ans);
